database/query/repositories.sql

```diff
@@ -45,6 +49,35 @@ WHERE project_id = $1 AND webhook_id IS NOT NULL
     AND (lower(provider) = lower(sqlc.narg('provider')::text) OR sqlc.narg('provider')::text IS NULL)
 ORDER BY repo_name;
 
+-- name: ListEligibleRepositoriesAfterID :many
+SELECT r.*
+FROM (
+  SELECT *
+  FROM repositories
+  WHERE repositories.id > $1
+  ORDER BY id
+  LIMIT sqlc.arg('limit')::bigint
+) r
+JOIN LATERAL (
```

Comment:
It looks like `LATERAL` converts this from a HashAggregate of a NestedLoop to a NestedLoop that runs an Aggregate inside:

```
EXPLAIN SELECT r.*
FROM (
  SELECT *
  FROM repositories
  ORDER BY id
  LIMIT 10
) r
JOIN LATERAL (
  SELECT MIN(rde.last_updated) AS min_last_updated
  FROM rule_evaluations re
  INNER JOIN rule_details_eval rde ON rde.rule_eval_id = re.id
  WHERE re.repository_id = r.id
) sub ON sub.min_last_updated + interval '1h' < NOW()
ORDER BY r.id;
                                                       QUERY PLAN                                                        
-------------------------------------------------------------------------------------------------------------------------
 Nested Loop  (cost=31.75..318.77 rows=10 width=338)
   ->  Limit  (cost=0.14..2.30 rows=10 width=338)
         ->  Index Scan using repositories_pkey on repositories  (cost=0.14..47.45 rows=220 width=338)
   ->  Aggregate  (cost=31.61..31.63 rows=1 width=8)
         Filter: ((min(rde.last_updated) + '01:00:00'::interval) < now())
         ->  Nested Loop  (cost=8.12..31.60 rows=5 width=8)
               ->  Bitmap Heap Scan on rule_evaluations re  (cost=7.97..15.08 rows=3 width=16)
                     Recheck Cond: (repository_id = repositories.id)
                     ->  Bitmap Index Scan on rule_evaluations_results_name_lower_idx  (cost=0.00..7.97 rows=3 width=0)
                           Index Cond: (repository_id = repositories.id)
               ->  Index Scan using idx_rule_detail_eval_ids on rule_details_eval rde  (cost=0.15..5.50 rows=1 width=24)
                     Index Cond: (rule_eval_id = re.id)
(12 rows)
```

vs

```
EXPLAIN SELECT r.id, r.provider, r.project_id, r.repo_owner, r.repo_name, r.repo_id, r.is_private, r.is_fork, r.webhook_id
FROM (
  SELECT *
  FROM repositories
  ORDER BY id
  LIMIT 10
) r
JOIN rule_evaluations AS re on re.repository_id = r.id
JOIN rule_details_eval rde ON rde.rule_eval_id = re.id
  WHERE re.repository_id = r.id
GROUP BY r.id, r.provider, r.project_id, r.repo_owner, r.repo_name, r.repo_id, r.is_private, r.is_fork, r.webhook_id
HAVING MIN(rde.last_updated) + interval '1h' < NOW()                                                                                                                             ORDER BY r.id;
                                                             QUERY PLAN                                                              
-------------------------------------------------------------------------------------------------------------------------------------
 Sort  (cost=28.33..28.36 rows=13 width=146)
   Sort Key: r.id
   ->  HashAggregate  (cost=27.39..28.09 rows=13 width=146)
         Group Key: r.id, r.provider, r.project_id, r.repo_owner, r.repo_name, r.repo_id, r.is_private, r.is_fork, r.webhook_id
         Filter: ((min(rde.last_updated) + '01:00:00'::interval) < now())
         ->  Nested Loop  (cost=2.67..26.39 rows=40 width=154)
               ->  Hash Join  (cost=2.52..19.79 rows=26 width=162)
                     Hash Cond: (re.repository_id = r.id)
                     ->  Seq Scan on rule_evaluations re  (cost=0.00..15.10 rows=510 width=32)
                     ->  Hash  (cost=2.40..2.40 rows=10 width=146)
                           ->  Subquery Scan on r  (cost=0.14..2.40 rows=10 width=146)
                                 ->  Limit  (cost=0.14..2.30 rows=10 width=338)
                                       ->  Index Scan using repositories_pkey on repositories  (cost=0.14..47.45 rows=220 width=338)
               ->  Index Scan using idx_rule_detail_eval_ids on rule_details_eval rde  (cost=0.15..0.25 rows=1 width=24)
                     Index Cond: (rule_eval_id = re.id)
(15 rows)
```

While there are more steps in the non-lateral query plan, it seems to plan and execute substantially faster than the LATERAL version.  In an environment with more rows, this is the difference between `EXPLAIN ANALYZE` taking 30ms and 2ms of execution time.